cyanure.py

```diff
@@ -364,9 +364,9 @@ def fit(self,X,y,lambd=0,lambd2=0,lambd3=0,solver='auto',tol=1e-3,it0=10,max_epo
 
         y=np.squeeze(y)
         uniq=np.unique(y)
-        if uniq.shape[0] != 2:
-            print(uniq.shape[0]+" classes detected; use MulticlassClassifier instead")
-            return
+        nb_classes = len(uniq)
+        if nb_classes != 2:
+            raise ValueError(f"{nb_classes} classes detected; use MulticlassClassifier instead")
```

Comment:
I use so-called f-strings here which needs Python 3.6+. I am guessing it is fine for this package, but let me know otherwise and I'll use `.format` instead.

Reply:
I checked around me and there are still some people using Python 3.5.  Not sure how to use .format, but it should be better if it does not cause compatibility issues.

Reply:
note that X, y = fetch_rcv1(return_X_y=True)  does not provide a dataset for binary classification. It is either multilabel of multiclass. 
The rcv1 dataset I was using can be found here 
http://pascal.inrialpes.fr/data2/mairal/data/

Reply:
> note that X, y = fetch_rcv1(return_X_y=True) does not provide a dataset for binary classification. It is either multilabel of multiclass.

Yep I figured that out afterwards, thanks!